vm/systemSmartContracts/staking.go

```diff
@@ -72,20 +64,37 @@ func NewStakingSmartContract(
 	if check.IfNil(args.Marshalizer) {
 		return nil, vm.ErrNilMarshalizer
 	}
+	if args.StakingSCConfig.WaitingNodesPercentage < 0 {
+		return nil, vm.ErrNegativeWaitingNodesPercentage
+	}
+	if args.StakingSCConfig.BleedPercentagePerRound < 0 {
+		return nil, vm.ErrNegativeBleedPercentagePerRound
+	}
+	if args.StakingSCConfig.MaximumPercentageToBleed < 0 {
+		return nil, vm.ErrNegativeMaximumPercentageToBleed
+	}
 
 	reg := &stakingSC{
-		minStakeValue:            big.NewInt(0).Set(args.MinStakeValue),
 		eei:                      args.Eei,
-		unBondPeriod:             args.UnBondPeriod,
+		unBondPeriod:             args.StakingSCConfig.UnBondPeriod,
 		stakeAccessAddr:          args.StakingAccessAddr,
 		jailAccessAddr:           args.JailAccessAddr,
-		numRoundsWithoutBleed:    args.NumRoundsWithoutBleed,
-		bleedPercentagePerRound:  args.BleedPercentagePerRound,
-		maximumPercentageToBleed: args.MaximumPercentageToBleed,
+		numRoundsWithoutBleed:    args.StakingSCConfig.NumRoundsWithoutBleed,
+		bleedPercentagePerRound:  args.StakingSCConfig.BleedPercentagePerRound,
+		maximumPercentageToBleed: args.StakingSCConfig.MaximumPercentageToBleed,
 		gasCost:                  args.GasCost,
-		minNumNodes:              int64(args.MinNumNodes),
+		minNumNodes:              args.MinNumNodes,
 		marshalizer:              args.Marshalizer,
 	}
+	ok := true
+	conversionBase := 10
+	reg.stakeValue, ok = big.NewInt(0).SetString(args.StakingSCConfig.GenesisNodePrice, conversionBase)
+	if !ok || reg.stakeValue.Cmp(zero) < 0 {
+		return nil, vm.ErrNegativeInitialStakeValue
+	}
+
+	reg.maxNumNodes = reg.minNumNodes + uint64(float64(reg.minNumNodes)*args.StakingSCConfig.WaitingNodesPercentage)
```

Comment:
This WaitingNodesPercentage name is misleading. When I first read the variable, I thought there is a waiting list for the nodes.  Why not using an integer value instead of percentage, it is more flexible to work with.

Reply:
done